test/utils/data_to_params_test.exs

```diff
@@ -34,6 +34,64 @@ defmodule JSONAPI.DataToParamsTest do
     }
   end
 
+  test "converts to many relationship" do
+    incoming = %{
+      "data" => %{
+        "id" => "1",
+        "type" => "user",
+        "attributes" => %{
+          "foo-bar" => true
+        },
+        "relationships" => %{
+          "baz" => %{
+            "data" => [
+              %{"id" => "2", "type" => "baz"},
+              %{"id" => "3", "type" => "baz"}
+            ]
+          }
+        }
+      }
+    }
+
+    result = JSONAPI.Utils.DataToParams.process(incoming)
+
+    assert result == %{
+      "id" => "1",
+      "type" => "user",
+      "foo-bar" => true,
+      "baz-id" => ["2", "3"]
+    }
+  end
+
+  test "converts polymorphic" do
+    incoming = %{
+      "data" => %{
+        "id" => "1",
+        "type" => "user",
+        "attributes" => %{
+          "foo-bar" => true
+        },
+        "relationships" => %{
+          "baz" => %{
+            "data" => [
+              %{"id" => "2", "type" => "baz"},
+              %{"id" => "3", "type" => "yooper"}
+            ]
+          }
+        }
+      }
+    }
+
+    result = JSONAPI.Utils.DataToParams.process(incoming)
+
+    assert result == %{
+      "id" => "1",
+      "type" => "user",
+      "foo-bar" => true,
+      "baz-id" => ["2", "3"]
```

Comment:
Doesn't this mean you have a bug? Imagine a system like this... I have a super class called "Car", and I want to create two records, a "Ford" and a "Subaru". The way I'm reading this you've lost the typing information that the user provided — thus in my scenario I wouldn't know that a user was creating a "Subaru" and a "Ford".

Reply:
Fixed!  Great point.  

https://github.com/jeregrine/jsonapi/pull/222/commits/2481ee93f0dbb30902401e8bc2493d23264cbc4d

https://github.com/jeregrine/jsonapi/pull/222/commits/e4ffe0643bc06f7cc11df9c4d3b059158ae6667e